### DISCUSSION TITLE
Shelly3EM in Opendtu on Battery einbinden

### DISCUSSION THREAD

**Discussion by Milchtrinker99** [hoylabs/OpenDTU-OnBattery#1034] [General]
Hallo,
das ist mein erster Post, seid bitte nicht so hart zu mir. :)
Ich weiß leider nicht wie ich den Shelly 3 EM einbiden kann,
Ich hatte vorher den Tasmota, und der lief einwandfrei.
Ich habe schon den Pfad eingegeben http://ip-adresse/rpc/Shelly.GetStatus, aber der funktioniert leider nicht.

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/162645372/49ea45b1-9961-4fe3-8007-28648f750544)

Den Befehl habe ich auch eingegeben, mit der Fehlermeldung.
 
Im HA sehe ich den, und da zeigt er mir auch die Daten an.
Kann mir einer weiterhelfen??

Vielen Dank schon mal .

LG

**Comment by genmad** [comment]
Hast du hier schon gelesen? 
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Power-Meter

**Comment by Solarteur** [comment]
Bei mir funktionieren diese Einstellungen mit einem Shelly 3EM:
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/117977338/52a5bf39-3f64-4e31-8994-749d19b5a840)

Wichtig: "total_power" musste ich eingeben, obwohl das schon im Feld "JSON path" voreingegeben war. Genau nach Wiki, wie @genmad schon schreibt.

**Comment by stegbth** [comment]
Hi @Milchtrinker99 ,

der Energiemesser hat mich auch ein bissl Zeit gekostet ;)
Ich habe einen Shelly Pro 3EM und der hat andere Aufrufparameter.

URL lautet:
http://192.168.xx.yy/rpc/EM.GetStatus?id=0
beim JSON Pfad steht: a_act_power
bzw. b_act_power bei Phase 2 usw.

vg
Thomas

**Reply by Andi1980** [reply]
Der Post ist zwar eine Weile her, aber der über alle 3 Phasen saldierte Wert ist:

URL: http://192.168.xxx.yyy/rpc/EM.GetStatus?id=0
JSON-Pfad: total_act_power 

Vielleicht hilfts mal jemandem.

**Reply by xmike13** [reply]
vielen Dank, ich werd’s bald brauchen

**Comment by Alex199608** [comment]
Guten Morgen😊 der beitrag ist schon eine weile her aber ich schaffe es nicht den shelly Pro3EM mit der opendtu zu verbinden. Ich habe es gestenr und heute schon vergeblich versucht.. ist der URL: http://<ipadressshelly>/rpc/Shelly.GetStstus veraltet? 
Bei JSON-Pfad habe ich <total_act_power> ei getragen aber es kommt immer der fehler dass die verbindung abgelehnt wurde🤔 bin jetzt leider gerade nicht vorort um den von @Andi1980  genannten URL zu probieren aber ist der aktuell?

LG Alex

**Comment by Andi1980** [comment]
Kurz zur Zusammenfassung: 

OpenDTU-onBattery Version: 2025.10.11
Shelly pro3em firmware: 1.7.1


Url get: http://192.168.xxx.xxx/rpc/EM.GetStatus?id=0
Keine Authentifizierung
Die URL 192.168.xxx.xxx ist die Netzwerkadresse deines Shellys in deinem Netzwerk.


Json: total_act_power
Einheit: W

Geht aktuell habe extra nachgeschaut.

**Reply by Alex199608** [reply]
Vielen dank! Hab die gleiche firmware und OPENDTU version👍 hast du im shelly noch irgendwas umgestellt oder läuft das dann von allein wenns in der opendtu konfiguriert wird? Ich probiers mittag gleich aus obs mit dem URL funktioniert, danke im vorraus 😊

**Comment by Alex199608** [comment]
Habs jetz ausprobiert aber es klappt trotzdem nicht😳 mach ich was falsch?
![IMG_0911](https://github.com/user-attachments/assets/25cce8cb-69a7-4172-88d8-2eb0e6d1d895)
![IMG_0912](https://github.com/user-attachments/assets/843eef87-dcaf-4f9a-b881-c1169bb71f2d)

**Reply by Sorbat** [reply]
Hast du mal den json Pfad, der im Wiki steht versucht?

**Reply by kaktus2910** [reply]
Den Shelly auch auf Port 1010 gestellt ?

**Reply by gitisgreat2023** [reply]
> Habs jetz ausprobiert aber es klappt trotzdem nicht😳 mach ich was falsch? ![IMG_0911](https://private-user-images.githubusercontent.com/244669492/515690646-25cce8cb-69a7-4172-88d8-2eb0e6d1d895.jpeg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM0Njk2NzUsIm5iZiI6MTc2MzQ2OTM3NSwicGF0aCI6Ii8yNDQ2Njk0OTIvNTE1NjkwNjQ2LTI1Y2NlOGNiLTY5YTctNDE3Mi04OGQ4LTJlYjBlNmQxZDg5NS5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMThUMTIzNjE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjVlZmUxYjZjMDI0OGY5NzJjM2JkOGY4MGNiYTVjMmFiMGU4MjMyMzk5Y2MyMWU1OWI3MmI1NDNlOWM0OWVkZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.KMgY_e9NiBr3MoDeHHeOxveURkaXlCTseLohke_uR5c) ![IMG_0912](https://private-user-images.githubusercontent.com/244669492/515690657-843eef87-dcaf-4f9a-b881-c1169bb71f2d.jpeg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM0Njk2NzUsIm5iZiI6MTc2MzQ2OTM3NSwicGF0aCI6Ii8yNDQ2Njk0OTIvNTE1NjkwNjU3LTg0M2VlZjg3LWRjYWYtNGY5YS1iODgxLWMxMTY5YmI3MWYyZC5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMThUMTIzNjE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGYzYmMyMTFkYzg1YTc5ZjFhMTEyMDBiMjNkNGQ4ZDA5YTZhNTllOGQyMTY0OTgyZTM3MGM0NmJhZTM1MDA4MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.-Z7-41MY5nBYmq4thdg0b49bLyJ3pNTn4RioqRqCxFQ)

Statt rpc etc nur /status

zB wenn er dtu heißt in der FRITZ!Box (sonst ip)
http://dtu.fritz.box/status

**Reply by Alex199608** [reply]
@Sorbat  welche genau meinst du? Hatte mit der alten URL auch total_power und /em:0/total_act_power  . Allerdings nocj mit der alten URL.

**Reply by Sorbat** [reply]
Welche alte url?

**Reply by Sorbat** [reply]
> Den Shelly auch auf Port 1010 gestellt ? 

Wofür soll das gut sein?

**Reply by Alex199608** [reply]
@kaktus2910 ja hatte ich schon probiert allerdings auch mit der alten URL.🤔 muss ich heute abend testen.

@gitisgreat2023 ne des hab ich noch nich probiert. Hab i a no nirgends gelesen aber dsnke für den tipp! Werds später probieren danke.

**Reply by Alex199608** [reply]
> Welche alte url?

http://ipdesshelly/rpc/Shelly.GetStatus

**Reply by Sorbat** [reply]
> Statt rpc etc nur /status

/status beim pro? Das wäre neu 🤔

**Reply by Sorbat** [reply]
> @Sorbat  welche genau meinst du? Hatte mit der alten URL auch total_power und /em:0/total_act_power  . Allerdings nocj mit der alten URL.

https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Power-Meter


Das steht bei mir

http://ip/rpc/Shelly.GetStatus

/em:0/total_act_power

**Reply by Alex199608** [reply]
> > @Sorbat  welche genau meinst du? Hatte mit der alten URL auch total_power und /em:0/total_act_power  . Allerdings nocj mit der alten URL.
> 
> https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Power-Meter
> 
> Das steht bei mir
> 
> http://ip/rpc/Shelly.GetStatus
> 
> /em:0/total_act_power

Hast du auch eine Pro? 
genau das was bei dir steht hab ich schon etliche male probiert.. Shelly auch schon zurückgesetzt und aktualisiert, eine feste IP im
Router vergeben etc. ich weis einfach nicht wo der fehler sein könnte..

**Reply by Sorbat** [reply]
Ja natürlich einen Pro.

Wie ist die genaue Modellbezeichnung von deinem Gerät?

Was siehst du, wenn du die URL im Browser aufrufst?

**Reply by Alex199608** [reply]
Oke. 
meinst du die bezeichnung aus der shelly app oder wie der shelly im router benannt ist?
Wenn ich die IP im browser eingib komm ich auf die seite vom shelly.
<img width="1170" height="2532" alt="IMG_0915" src="https://github.com/user-attachments/assets/1577c8de-34c5-44ea-a72c-e16103285c71" />

**Reply by kaktus2910** [reply]
Ich war wieder zu schnell . Was ich meinte mit Port 1010 ist für einen Akku gewesen.

**Reply by Sorbat** [reply]
> meinst du die bezeichnung aus der shelly app oder wie der shelly im router benannt ist?

Weder noch. Die genaue Modellbezeichnung von deinem Gerät. 

> Wenn ich die IP im browser eingib komm ich auf die seite vom shelly.

Und jetzt nochmal versuchen mit der URL

**Reply by Alex199608** [reply]
> > meinst du die bezeichnung aus der shelly app oder wie der shelly im router benannt ist?
> 
> Weder noch. Die genaue Modellbezeichnung von deinem Gerät.
> 
> > Wenn ich die IP im browser eingib komm ich auf die seite vom shelly.
> 
> Und jetzt nochmal versuchen mit der URL

Weis jetz nicht was du meinst🤔 wo soll die stehen? Also du meinst schon den shelly oder die dtu?
Okey probiere ich hernach aus!

**Reply by Alex199608** [reply]
Ich kann jetzt weder die IP noch den URL vom shelly öffnen😳

**Reply by Andi1980** [reply]
Zum einen, wenn DU den Shelly mit dem Webbrowser nicht erreichst, kann es die DTU auch nicht.
Zum anderen würde ich die Geräte-ID etc. in Foren Posts schwärzen.

**Reply by gitisgreat2023** [reply]
@Alex199608 
Dein Shelly ist nicht erreichbar mittels Browser? Das ist schlecht...
Sicher dass Ethernet enabled ist?
Geh mal mittels den App mittels bluetooth auf's Gerät. Dann ethernet einschalten, damit er übers LAN erreichbar wird...
Beim Fritz!Box: dein shelly suchen und dann fixen IP address zuweisen und einen hostname, zB shelly.
Dann egal welche IP es ist (dyn auch) ist er über shelly.fritz.box erreichbar.

Dann mal firmware upgrade auf'm Shelly machen. Settings -> Device Settings -> Firmware

Ich habe auch einen Shelly Pro 3EM.
Dann ist der URL: (statt shelly dann natürlich dein hostname oder IP address)

http://shelly.fritz.box/rpc/Shelly.GetStatus

Ich habe es mit einem pw
Also digest
admin
und den pw natürlich

JSON field: /em:0/total_act_power

**Reply by gitisgreat2023** [reply]
> > Statt rpc etc nur /status
> 
> /status beim pro? Das wäre neu 🤔

stimmt, beim pro ist es 
http://shelly.fritz.box/rpc/Shelly.GetStatus
/em:0/total_act_power

**Reply by Alex199608** [reply]
Mein Shelly hängt nur im WLAN. Ich kann pber die app mittels wlan auf de shelly zugreifen und hab heute auch das update der firmware gemacht. Ich konnte heute auch schon mittels browser und IP auf den shelly zugreifen aber jetzt komm ich einfach nicht drauf. Daten liefert er ganz normal an die app. Werksreset lässt er mich nicht machen, schlägt immer fehl.  Das mit der festen IP mache ich hernach mal im router.
Ich habe eine Vodafone router (kabelinternet)

**Comment by Alex199608** [comment]
Guten Morgen, es gibt neue erkentnisse. 
Nach abklemmen des shellys vom strom kann ich nun über die ip und den URL zugreifen auf de. Shellly. Im router war UPnp desktiviert, das hab ich jetz altiviert. Hab dann nochmal alles neugestartet aber es kommt immer der selbe fehler bei der verbindung. Hat noch einer eine idee?
![IMG_0924](https://github.com/user-attachments/assets/fbf0736f-ac43-4fdc-b12e-cc1c008799d9)
die verbindu
![IMG_0922](https://github.com/user-attachments/assets/51538670-e405-486e-9bbd-5398944d5288)
![IMG_0923](https://github.com/user-attachments/assets/2dc686dd-4f93-444d-a7d3-fe29d3026db1)

![IMG_0921](https://github.com/user-attachments/assets/6764b95b-4bee-4149-8037-71bc293fbddc)

**Reply by gitisgreat2023** [reply]
/em:0/total_act_power ist richtig, man sieht es ja in den Screenshots… komisch 
Welche dtu Version ist installiert?

**Reply by Alex199608** [reply]
Ich könnte es nocj versuchen die einzelnen phasen abzufragen🤔
Habe die onbattery 2025.10.11

**Reply by gitisgreat2023** [reply]
Was du noch machen kannst:
wenn du in der DTU es abfragst, kannst du auf dem Shelly sehen ob die Anfrage durchkommt.
Das findest du unter diagnostics.
Da sieht man jede Anfrage...

Einfach in der DTU die Testanfrage abgeschicken und du siehst ob er ankommt...

**Reply by Alex199608** [reply]
Hatte ich heute früh schonmal auf aber hatte nichts gesehen wenn ich die anfrage schicke. 
Habs jetz gerade nochmal versucht per IP oder url auf den shelly zuzugreifen aber funktioniert wieder nicht🤦‍♂️ immer wenn ich ihn vom strom nehme gehts eine zeit und dann nichtmehr.

**Reply by Andi1980** [reply]
Dann ist das wohl eher eine Frage für den Shelly Support, warum der sich verabschiedet.

**Reply by gitisgreat2023** [reply]
Ja sieht aus wie ein Shelly issue. Kannst du es mal am LAN hängen?

**Reply by Alex199608** [reply]
> Ja sieht aus wie ein Shelly issue. Kannst du es mal am LAN hängen?

Ja ich versuchs, brauch ein ewig langes kabel🤦‍♂️

**Comment by Sorbat** [comment]
Hattest du mal 

/em:0/total_act_power

bei json Pfad probiert?

**Reply by Alex199608** [reply]
> Hattest du mal
> 
> /em:0/total_act_power
> 
> bei json Pfad probiert?

Ja das habe ich im json ofad hinterlegt siehe fotos

**Reply by Sorbat** [reply]
Ich hatte auf deinen Fotos nur 

em:0/total_act_power

gesehen. Daher die Frage. Aber wenn es nun funktioniert ist ja gut

**Comment by Alex199608** [comment]
Fehler gefunden! Bin durch zufall drsuf gestoßen..bluetooth war deaktiviert..aber un RPC benutzen zu können muss bluetooth an sein🤦‍♂️
<img width="1170" height="2532" alt="IMG_0952" src="https://github.com/user-attachments/assets/6fa20f5a-8d49-40fe-968c-b0c1840801ad" />

**Reply by gitisgreat2023** [reply]
Ah super... danach kann man bluetooth wieder ausschalten... habe RPC aktiv und bluetooth aus.

Verstehe nicht warum die http call dann Ergebnisse gezeigt hat... screens hattest du ja gepostet.

**Comment by Alex199608** [comment]
Ja genau…
Ja versteh ich auch nicht ganz😂 aber die nulleinspeisung läuft jetz einwandfrei!

**Comment by Alex199608** [comment]
Guten Morgen :) ich hole das Thema mal wieder hoch..
habe gestern die OPEN DTU ON BATTERY auf die aktuellste version upgedatet und jetzt hab ich keine daten mehr vom PRO3EM..
konfiguration hatte ich vorher gespeichert und es ist auch alles so eingetragen wie vorher aber die http anfrage geht nicht durch. Über die URL kann ich den PRO3em auslesen aber die verbindung über die opendtu klappt nicht.. hat jemand eine idee?
[iCloud Fotos (2).zip](https://github.com/user-attachments/files/27580112/iCloud.Fotos.2.zip)
[iCloud Fotos (1).zip](https://github.com/user-attachments/files/27580114/iCloud.Fotos.1.zip)
[iCloud Fotos.zip](https://github.com/user-attachments/files/27580120/iCloud.Fotos.zip)

**Reply by stegbth** [reply]
> Guten Morgen :) ich hole das Thema mal wieder hoch.. habe gestern die OPEN DTU ON BATTERY auf die aktuellste version upgedatet und jetzt hab ich keine daten mehr vom PRO3EM.. konfiguration hatte ich vorher gespeichert und es ist auch alles so eingetragen wie vorher aber die http anfrage geht nicht durch. Über die URL kann ich den PRO3em auslesen aber die verbindung über die opendtu klappt nicht.. hat jemand eine idee? [iCloud Fotos (2).zip](https://github.com/user-attachments/files/27580112/iCloud.Fotos.2.zip) [iCloud Fotos (1).zip](https://github.com/user-attachments/files/27580114/iCloud.Fotos.1.zip) [iCloud Fotos.zip](https://github.com/user-attachments/files/27580120/iCloud.Fotos.zip)

Hi @Alex199608,

das Upgrade auf OpenDTU-onBattery 2026.03.31 habe am Wochenende durchgefuehrt.
Der Shelly 3EM Pro laeuft auf Version 1.7.4.
Die Aufruf URL im OpenDTUoB lautet: http://192.168.0.32/rpc/EM.GetStatus?id=0 und genutzter JSON-Pfad ist total_act_power.

Der Netzbezug wird in der OpenDTU entsprechend angezeigt.

Ich bin mir bei deinem JSON-Pfad nicht sicher, der sieht zumindest anders aus, als bei mir.

viele Gruesse
Tom

p.s. wobei ich oben sehe, dass Du das auch so eingetragen hattest.

**Reply by Alex199608** [reply]
Guten Morgen, danke für die rückmeldung ;)
habe es mal mit deinem pfad probiert aber funktioniert auch nicht..
Vorher hatte mit meinem pfad alles funktioniert und ich kann auch mit dem URL die daten des pro3Em auslesen aber doe verbindung klappt nicht🤔 mein shelly hat schon 1.7.5🤔🤔

**Reply by stegbth** [reply]
> Guten Morgen, danke für die rückmeldung ;) habe es mal mit deinem pfad probiert aber funktioniert auch nicht.. Vorher hatte mit meinem pfad alles funktioniert und ich kann auch mit dem URL die daten des pro3Em auslesen aber doe verbindung klappt nicht🤔 mein shelly hat schon 1.7.5🤔🤔

Hi @Alex199608,
jetzt werfe ich einfach mal ein paar Netzwerkfragen rein ;)

1. wie ist der Shelly und dein OpenDTUoB mit dem Netzwerk verbunden? (Wlan oder Netzwerk)
2. welche internen IP Adressen haben die beiden?
3. falls beide Geraete via Wlan verbunden sind, kann es sein, dass die Kommunikation zwischen Wlan Geraeten am Router unterbunden ist?
4. ist es moeglich den Shelly - zumindest temporaer - mittels Netzwerkkabel an den Router zu verbinden?
 (Vorsicht: der Shelly hat dann auf dem Netzwerkport eine ANDERE IP, als am Wlan.)
 
 vg
 Thomas

**Comment by Alex199608** [comment]
@stegbth es sind beide via wlan angebunden. War vor dem update genauso und hat geklappt! Hatte bei der ersten eineichtunf genau das selbe problem weil 1. die kommunikation im router deaktiviert war und RPC beim shelly deaktiviert war. 
IP Dtu: 192.186.0.42
Ip Shelly:192.168.0.113
Kann den shelly leider nicht per lan verbinden..

**Comment by Andi1980** [comment]
Hast Du es mal mit einem reset des Shellys probiert. Ich hatte mal ein ähnliches Problem, das ich den nicht mehr im Browser aufrufen konnte egal was ich machte aber Homeassistant ging. Irgendwann habe ich den Shelly reseted und alles ging wieder. War glaube ich auch nach einem Update. Vlt. irgendwelche Reste im Speicher...

Grüße
Andi

**Comment by Alex199608** [comment]
Servus, habe alles resetet und shelly stromlos gemacht…nach tagelangem suchen war es das allereinfachste😂 nachdem ich gesehen hab dass mein victronsystem den shelly auch nichtmehr anzeigt hab ich meinen router neugestartet und schon geht alles.. danke für jede antwort 🙏

**Comment by andil3190** [comment]
Hallo, habe mit meinem Shelly pro 3EM immer wieder Sendeabbrüche ( WLAN Signal grenzwertig ) und möchte über einen Shelly Plug M Gen.3 ( mit range extender ) das Signal Richtung Router vertärken. Wie sieht da die Konfiguration bzw. der Pfad aus ( URL+JSON )?
Danke im Voraus
Schöne Grüße
Andi

**Reply by SW-Niko** [reply]
Hallo @andil3190,
mit dem Range Extender habe ich keine Erfahrung. Sorry.
Aber ... das hast du schon gesehen? PR: #2608 
Damit kannst du einfach überprüfen, ob die WLAN Verbindung ausreichend gut ist.

_powerMeter: [HTTP/JSON] Poll/Http errors: 0.1% [6 of 5844 polls]_

**Comment by Andi1980** [comment]
> Hallo, habe mit meinem Shelly pro 3EM immer wieder Sendeabbrüche ( WLAN Signal grenzwertig ) und möchte über einen Shelly Plug M Gen.3 ( mit range extender ) das Signal Richtung Router vertärken. Wie sieht da die Konfiguration bzw. der Pfad aus ( URL+JSON )?
> Danke im Voraus
> Schöne Grüße
> Andi

Was hat der Range Extender / Access-Point Modus denn mit der JSON Abfrage der Energiedaten zu tun?

Das eine ist eine Setup Geschichte und das andere eine Datenabfrage von einem andern Gerät im Netzwerk. Klar dazu muss der Shelly erreichbar sein aber sonst?...
